org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -462,6 +488,249 @@ private void appendGuessingCompletion(StringBuilder buffer, CompletionProposal p
 		}
 	}
 
+	private String[] guessParameters(char[][] parameterNames, CompletionProposal proposal) throws JavaModelException {
```

Comment:
can you please keep all those copied methods from ParameterGuesser in a forked ParameterGuesser file?